master/internal/api/params.go

```diff
@@ -0,0 +1,35 @@
+package api
+
+import "math"
+
+// EffectiveOffset returns effective offset.
+func EffectiveOffset(reqOffset int, total int) (offset int) {
+	switch {
+	case reqOffset < -total:
+		return 0
+	case reqOffset < 0:
+		return int(math.Max(float64(total+reqOffset), 0))
+	default:
+		return reqOffset
+	}
+}
+
+// EffectiveLimit returns effective limit.
```

Comment:
I understand that this comment is required for exported functions, but if we have to have it, maybe explain what "effective" means?